src/intents.js

```diff
@@ -0,0 +1,61 @@
+const intentClass = 'coz-intent'
+
+function getUrl (intent, temporaryMockedIntentUrl) {
+  return Promise.resolve(temporaryMockedIntentUrl)
+}
+
+function injectService (url, element, intent) {
+  const document = element.ownerDocument
+  if (!document) throw new Error('Cannot retrieve document object from given element')
+
+  const window = document.defaultView
+  if (!window) throw new Error('Cannot retrieve window object from document')
+
+  const iframe = document.createElement('iframe')
+  iframe.setAttribute('src', url)
+  iframe.classList.add(intentClass)
+  element.appendChild(iframe)
+
+  // Keeps only http://domain:port/
+  const serviceOrigin = url.split('/', 3).join('/')
+
+  return new Promise((resolve, reject) => {
+    let handshaken = false
+    const messageHandler = (event) => {
+      if (event.origin !== serviceOrigin) return
+
+      if (event.data === 'intent:ready') {
+        handshaken = true
+        return event.source.postMessage(intent.data, event.origin)
+      }
+
+      window.removeEventListener('message', messageHandler)
+
+      if (event.data === 'intent:error') {
+        return reject(new Error('Intent error'))
+      }
+
+      return handshaken
+        ? iframe.parentNode.removeChild(iframe) && resolve(event.data)
+        : reject(new Error('Unexpected handshake message from intent service'))
+    }
+
+    window.addEventListener('message', messageHandler)
+  })
+}
+
+export function start (client, intent, element, temporaryMockedIntentUrl) {
+  ['action', 'type'].forEach((property) => {
```

Comment:
Oh, *that part*. Well, creating the intent in the DB and getting the service URL is really one and the same step, so I don't think there will be more than one function. But yeah, our first proposal makes more and more sense.
Do you want to change the current PR to include it, or would you rather sit on it for a while and maybe do it later? I'm totally fine with both.